hr_expense_cancel/models/hr_expense.py

```diff
@@ -0,0 +1,63 @@
+# Copyright 2019 Tecnativa - Ernesto Tejeda
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class HrExpenseSheet(models.Model):
+    _inherit = "hr.expense.sheet"
+
+    @api.multi
+    def action_cancel(self):
+        for sheet in self:
+            account_move = sheet.account_move_id
+            sheet.account_move_id = False
+
+            # If the sheet is paid then remove payments
+            if sheet.state == 'done':
+                payments = self.env['account.payment'].search([
+                    ('expense_sheet_id', '=', sheet.id),
+                    ('state', '!=', 'cancelled'),
+                ])
+                if sheet.expense_line_ids[0].payment_mode == 'own_account':
+                    self._remove_move_reconcile(payments, account_move)
+                    self._cancel_payments(payments)
+                else:
+                    # In this case, during the cancellation the journal entry
+                    # will be deleted
+                    self._cancel_payments(payments)
+                    payments.unlink()
+
```

Comment:
Don't put empty lines inside a method: put a comment explaining the new "section" or simply put it together.

Reply:
done